### PR TITLE
APP-2072 Move prime to packages/legacy in a pnpm workspace.

### PR DESCRIPTION
**Pull request by ethanlook**
# Overview

This PR restructures PRIME to use [pnpm workspaces][] according to the proposal in [this wiki][]. The `@viamrobotics/prime` package will remain stable for now during our migration to the new structure. This package will be removed once we've completed the move and applications no longer depend on `@viamrobotics/prime`.

[pnpm workspaces]: https://pnpm.io/workspaces
[this wiki]: https://viam.atlassian.net/l/cp/cM6GgWX5

# Test Plan

- Verified existing commands work as expected (playground, storybook, etc)

# Changelog

- Added `pnpm-workspace.yaml`
- Moved package under `packages/legacy`
    - Fixed all path resolution
- Updated READMEs, added new top-level README
- Updated Github Workflows

# Review requests

Please pull down this branch and test your normal workflows with PRIME and make sure they work as expected.

### CODE REVIEW

**Review by mrloureed — APPROVED** [review]
LGTM 👏

**Comment by DTCurrie** [comment]
I can't run `pnpm build` in `packages/legacy` and I am seeing this with `pnpm storybook`:

Screenshot 2023-07-24 at 10 43 58 AM

**Comment by ethanlook** [comment]
> I can't run `pnpm build` in `packages/legacy` and I am seeing this with `pnpm storybook`:

@DTCurrie Can you try again? Did you `pnpm install`?

File: .gitignore [hunk 1]
```diff
@@ -25,5 +24,5 @@ prime
 *.sln
 *.sw?
 /test-results/
-/playwright-report/
-/playwright/.cache/
+**/playwright-report/
+**/playwright/.cache/
```
Comment (mcous) [inline .gitignore:28]:
```suggestion
playwright-report
playwright/.cache
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -70,7 +70,7 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: npm-dist
-          path: dist
+          path: packages/legacy/dist
       - name: Publish 🚀
         uses: JS-DevTools/npm-publish@541aa6b21b4a1e9990c95a92c21adc16b35e9551
         with:
```
Comment (mcous) [inline .github/workflows/ci.yml:76]: We'll need to add the [`package` input](https://github.com/JS-DevTools/npm-publish#usage-1) to `with` to make sure we publish from the right place:

```yaml
with:
  token: ${{ secrets.NPM_TOKEN }}
  package: packages/legacy
```

While testing, you can also try things out with the `dry-run` input to validate the correct files are being packed and published

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -70,7 +70,7 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: npm-dist
-          path: dist
+          path: packages/legacy/dist
       - name: Publish 🚀
         uses: JS-DevTools/npm-publish@541aa6b21b4a1e9990c95a92c21adc16b35e9551
```
Comment (mcous) [inline .github/workflows/ci.yml:75]: Could bump this to latest while we're here

```suggestion
        uses: JS-DevTools/npm-publish@5a85faf05d2ade2d5b6682bfe5359915d5159c6c # v2.2.1
```

File: package.json [hunk 4]
```diff
@@ -1,101 +1,10 @@
...
-    "svelte": "^4.0.5",
-    "svelte-check": "^3.4.3",
-    "svelte-preprocess": "^5.0.3",
-    "tailwindcss": "^3.3.1",
-    "typescript": "^5.1.6",
-    "vite": "4.3.1",
-    "vue": "^3.2.45",
-    "vue-tsc": "^1.2.0"
-  }
+  "packageManager": "pnpm@8.6.3"
```
Comment (mcous) [inline package.json:9]: Could bump this to latest while we're here

```suggestion
  "packageManager": "pnpm@8.6.10"
```

File: packages/legacy/package.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  "main": "./dist/prime.umd.cjs",
+  "module": "./dist/prime.js",
+  "types": "./dist/main.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/prime.js",
+      "require": "./dist/prime.umd.cjs"
+    },
+    "./theme": {
+      "import": "./dist/theme.js"
```
Comment (mcous) [inline packages/legacy/package.json:18]:
```suggestion
      "import": "./theme.js"
```

File: packages/legacy/package.json [hunk 6]
```diff
@@ -0,0 +1,102 @@
+{
+  "name": "@viamrobotics/prime",
+  "version": "0.4.0",
+  "license": "Apache-2.0",
+  "type": "module",
+  "files": [
+    "dist"
```
Comment (mcous) [inline packages/legacy/package.json:7]:
```suggestion
    "dist",
    "theme.js"
```

**Review by mcous — CHANGES_REQUESTED** [review]
Re-org looks great! I think these should be all the changes we need to get publishing working with the new structure.

While testing, you can use the `dry-run` input on the publish step (along with messing with the `if`) to verify the correct stuff will get published

File: package.json [hunk 7]
```diff
@@ -1,101 +1,10 @@
...
-    "check-format": "pnpm run _prettier --check",
-    "format": "pnpm run _prettier --write",
-    "_prettier": "prettier \"./**/*.@(js|cjs|ts|svelte|md|mdx|yml|json)\"",
-    "test": "playwright test",
-    "test-dev": "playwright test --debug",
-    "storybook": "storybook dev -p 6006",
-    "storybook-docs": "storybook dev --docs --no-manager-cache",
-    "build-storybook": "storybook build --docs -o prime"
-  },
-  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (ethanlook) [reply package.json:44]: While I typically agree, the `legacy` package uses some old versions of packages and upgrades aren't straightforward. I want to add the latest version for the new packages (`core` and `storybook`) but moving deps to the root makes that challenging. Ok if I skip for now and we can move the deps to the root after this package has been removed?

[the remark this answers is not in the document]
Comment (mcous) [reply package.json:44]: Oh gotcha, that makes sense. Carry on!

File: .npmignore
[the remark this answers is not in the document]
Comment (ethanlook) [reply .npmignore:1]: Removed.

File: package.json [hunk 8]
```diff
@@ -1,101 +1,10 @@
 {
-  "name": "@viamrobotics/prime",
-  "version": "0.3.6",
+  "name": "prime",
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (ethanlook) [reply package.json]: Removed.

File: package.json [hunk 9]
```diff
@@ -1,101 +1,10 @@
 {
-  "name": "@viamrobotics/prime",
-  "version": "0.3.6",
+  "name": "prime",
```
[the remark this answers is not in the document]
Comment (ethanlook) [reply package.json]: Great catch, changed.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ethanlook) [reply .github/workflows/ci.yml]: Updated, thanks for the suggestion

File: package.json (hunk 4 above)
Comment (ethanlook) [reply package.json]: Updated, thanks for the suggestion

File: packages/legacy/package.json (hunk 5 above)
Comment (ethanlook) [reply packages/legacy/package.json]: Great catch, updated

File: packages/legacy/package.json (hunk 6 above)
Comment (ethanlook) [reply packages/legacy/package.json]: Great catch, added

**Comment by ethanlook** [comment]
> While testing, you can use the `dry-run` input on the publish step (along with messing with the `if`) to verify the correct stuff will get published

I validated with `dry-run` that the correct files would get published. Thank you for the recommendation!

**Pull request merged**